### ISSUE TITLE
Provider doesn't work on non-FHS systems (e.g. NixOS)

### ISSUE THREAD

**Issue by dsx** [Backblaze/terraform-provider-b2#25]
`libutil.so.1` is provided by glibc, however on non-FHS systems like NixOS its location is not under `/lib` hierarchy. Since provider is just a wrapper around the Python code, would it be possible to provide it with a path to the system Python (or whole environment with Backblaze Python library) instead?

```
╷
│ Error: [530919] Error loading Python lib '/run/user/1000/_MEIZgzHIw/libpython3.9.so.1.0': dlopen: libutil.so.1: cannot open shared object file: No such file or directory
│ 
│ 
│   with module.b2.b2_application_key.gitlab-runner-cache-key,
│   on modules/b2/main.tf line 5, in resource "b2_application_key" "gitlab-runner-cache-key":
│    5: resource "b2_application_key" "gitlab-runner-cache-key" {
│ 
╵
╷
│ Error: [530915] Error loading Python lib '/run/user/1000/_MEIIUBRQm/libpython3.9.so.1.0': dlopen: libutil.so.1: cannot open shared object file: No such file or directory
│ 
│ 
│   with module.b2.b2_bucket.gitlab-runner-cache-bucket,
│   on modules/b2/main.tf line 10, in resource "b2_bucket" "gitlab-runner-cache-bucket":
│   10: resource "b2_bucket" "gitlab-runner-cache-bucket" {
│ 
```

**Comment by ppolewicz** [comment]
@dsx please confirm whether you are using the newest release of the terraform provider?

@wbancer-reef I think you were the last one to work on this. I thought our binary is now totally static and doesn't rely on anything, yet here it seems that it does. Can we make it "more static" so that environment problems like this can't happen?

**Comment by dsx** [comment]
@ppolewicz Yes, that's the latest release that was available to be downloaded by `terraform init`.

As for making binaries «more static», I don't think it's a good idea to carry around whole glibc runtime. That would unnecessary inflate binary size for everybody only to fix some weird corner cases like non-FHS systems. On the other hand, creating a hermetic language environments is one of the strong selling points of e.g. NixOS. There are built-in [tools](https://nixos.org/manual/nixpkgs/stable/#python) that would make it trivial to provide a specific version of Python with all dependencies required to run Backblaze code as long as all parts and pieces loosely adhere to the conformant way of shipping Python package distribution. Think `virtualenv` on steroids.

Either way, I'd be happy to help with testing this on NixOS.

**Comment by ppolewicz** [comment]
I think we are now capable of running the provider on MUSL systems so tbh I'm a bit surprised that a glibc dependency is still causing a problem.

I don't think anyone cares about the binary size. Typical terraform provider is like 50KB and ours is like 24MB, so maybe it will take two seconds instead of one second for the CI server to download it.

**Comment by mlech-reef** [comment]
@dsx this is the same problem as in #26. The fix is being implemented.

**Comment by mszumocki-reef** [comment]
Fix will go out with 0.7 release.
